op-challenger/game/fault/preimages/large.go

```diff
@@ -54,41 +56,40 @@ func NewLargePreimageUploader(logger log.Logger, txMgr txmgr.TxManager, contract
 }
 
 func (p *LargePreimageUploader) UploadPreimage(ctx context.Context, parent uint64, data *types.PreimageOracleData) error {
-	// Split the preimage data into chunks of size [MaxChunkSize] (except the last chunk).
-	stateMatrix := matrix.NewStateMatrix()
-	chunk := make([]byte, 0, MaxChunkSize)
-	calls := []Chunk{}
-	commitments := make([][32]byte, 0, MaxLeafsPerChunk)
-	in := bytes.NewReader(data.OracleData)
-	for i := 0; ; i++ {
-		// Absorb the next preimage chunk leaf and run the keccak permutation.
-		leaf, err := stateMatrix.AbsorbNextLeaf(in)
-		chunk = append(chunk, leaf...)
-		commitments = append(commitments, stateMatrix.StateCommitment())
-		// SAFETY: the last leaf will always return an [io.EOF] error from [AbsorbNextLeaf].
-		if errors.Is(err, io.EOF) {
-			calls = append(calls, Chunk{chunk, commitments[:], true})
-			break
-		}
-		if err != nil {
-			return fmt.Errorf("failed to absorb leaf: %w", err)
-		}
-
-		// Only create a call if the chunk is full.
-		if len(chunk) >= MaxChunkSize {
-			calls = append(calls, Chunk{chunk, commitments[:], false})
-			chunk = make([]byte, 0, MaxChunkSize)
-			commitments = make([][32]byte, 0, MaxLeafsPerChunk)
-		}
+	chunks, err := p.splitChunks(data)
+	if err != nil {
+		return fmt.Errorf("failed to split preimage into chunks: %w", err)
 	}
 
 	uuid := p.newUUID(data)
-	err := p.initLargePreimage(ctx, uuid, data.OracleOffset, uint32(len(data.OracleData)))
+
+	// Fetch the current metadata for this preimage data, if it exists.
+	metadata, err := p.contract.GetProposalMetadata(ctx, batching.BlockLatest, []gameTypes.LargePreimageIdent{{Claimant: p.txMgr.From(), UUID: uuid}})
```

Comment:
```suggestion
	metadata, err := p.contract.GetProposalMetadata(ctx, batching.BlockLatest, gameTypes.LargePreimageIdent{Claimant: p.txMgr.From(), UUID: uuid})
```

Not sure I got that change exactly right but the varargs makes the call slightly nicer.